swagger_parser/lib/src/generator/generator.dart

```diff
@@ -16,126 +16,106 @@ import 'models/universal_rest_client.dart';
 /// Can be provided with arguments
 /// to specify custom path to yaml config.
 final class Generator {
-  /// Applies parameters set from yaml config file
+  /// Applies parameters directly from constructor
   /// and sets them to default if not found
-  Generator.fromYamlConfig(List<String> arguments) {
-    final yamlConfig = YamlConfig(arguments);
+  Generator({
+    required String schemaContent,
+    required String outputDirectory,
+    required ProgrammingLanguage? language,
```

Comment:
Why not to do 
```ProgrammingLanguage language = ProgrammingLanguage.dart```
with all fields, instead of doing it after the constructor?

Reply:
All arguments are marked as required for the same reasons I described above, 
This constructor is used by the Generator.fromYamlConfig factory constructor, which because of the mandatory nullable arguments can pass parameters directly from yamlConfig without checking each parameter for null, and each null parameter passed will be replaced with the default parameter.

Also, since swagger_parser_pages uses this constructor, it's a good idea to make sure we didn't miss any of the parameters when passing them.

Also, swagger_parser_pages can pass null to some fields, such as clientPostfix, so we need to set default values in the initializers to avoid setting them on the swagger_parser_pages side.